docs/localization/LOCALIZATION.md

```diff
@@ -74,343 +78,443 @@ export function localizationConfigurationFactory() {
+
+### Localization bundle
+
+#### How to generate a component bundle
+
+```shell
+ng run project-name:i18n
+```
 
-By default, a container needs to provide an override of translation keys to the presenter:
+It will create an `i18n` folder in the component folder, with an `en-GB.json` file containing the mapping between the localization key and the associated default value.
+
+Check the list of options available by running:
+```shell
+ng run project-name:i18n --help
+```
+
+#### How to generate an application bundle
+
+```shell
+ng run project-name:generate-translations
+```
+It will create the application bundle for each supported language.
+
+Check the list of options available by running:
+
+```shell
+ng run project-name:generate-translations --help
+```
+
+## Additional features
+
+### How to localize a date, decimal and currency
+
+Use Angular's built-in [DatePipe](https://angular.io/api/common/DatePipe), [DecimalPipe](https://angular.io/api/common/DecimalPipe) and [CurrencyPipe](https://angular.io/api/common/CurrencyPipe) pass the current locale as the last parameter.
+The locale is read from `this.localizationService.getCurrentLanguage()`. To be able to use this, your component must inject the `LocalizationService` dependency as a parameter of the constructor.
 
 ```typescript
-import {MyPresTranslation} from '../presenter/my.translation.ts';
+import { Component, Input } from '@angular/core';
+import { Localization, LocalizationService, Translatable } from '@o3r/localization';
+import { SimpleHeaderPresTranslation, translations } from './simple-header-pres.translation';
 
 @Component({
-  selector: 'o3r-forms-poc-cont',
-  templateUrl: './forms-poc-cont.template.html',
-  changeDetection: ChangeDetectionStrategy.OnPush
+  selector: 'o3r-simple-header-pres',
+  styleUrls: ['./simple-header-pres.style.scss'],
+  templateUrl: './simple-header-pres.template.html'
 })
-export class MyContComponent implements FormsPocContContext, Block {
+export class SimpleHeaderPresComponent implements Translatable<SimpleHeaderPresTranslation>, ... {
 
-  /**
-   * Localization of the component
-   */
-  @Input()
-  public translations: Partial<MyPresTranslation>;
+  // ...
+  public currentLanguage: string;
 
-  ...
+  constructor(public localizationService: LocalizationService) {
+    this.currentLanguage = localizationService.getCurrentLanguage();
+    // ...
+  }
 
-  public getMyPresContext(overrideContext) {
-    return {
-      translations,
-      inputs: {
-        // ...
-      }
-    };
+  /**
+   * Called upon language change to set current language
+   * @param  language
+   */
+  public useLanguage(language: string) {
+    this.localizationService.useLanguage(language);
   }
 }
 ```
 
-A container can have its own translation (for error messages for example). In this case the container become `Translatable` as following:
+Now we can start using pipes:
+
+```html
+<p>{{today | date: 'fullDate' : '' : currentLanguage}}</p>
```

Comment:
We are not forced to pass the currentLanguage
```html
<p>{{today | date: 'fullDate' }}</p>

<p>{{1.5487 | number}}</p>

<p>{{1.5487 | currency: 'EUR' : 'symbol'}}</p>
```
but if you want to and if you apply the fix I proposed above
```html
<p>{{today | date: 'fullDate' : '' : currentLanguage()}}</p>

<p>{{1.5487 | number: '' : currentLanguage()}}</p>

<p>{{1.5487 | currency: 'EUR' : 'symbol' : '' : currentLanguage()}}</p>
```